gen/emit/EmitJulia.cpp

```diff
@@ -70,47 +70,91 @@ struct EmitJulia : public Emitter
         }
     }
 
+    string topLevelPackage(const string& package) {
+        auto packageParts = StringUtil::split(package, '.');
+        if (packageParts.size() > 0) {
+            return packageParts.at(0);
+        } else {
+            return "";
+        }
+    }
+
     void emitDependencies()
     {
-        unordered_set<string> dependencies;
+        unordered_set<string> imports;
+        string lsPackage = topLevelPackage(ls.structname.package);
+        if (lsPackage != "") {
+            imports.insert("import " + lsPackage);
+        }
+
         for (auto& lm : ls.members) {
             auto& tn = lm.type.fullname;
             if (!ZCMGen::isPrimitiveType(tn) &&
-                dependencies.find(tn) == dependencies.end() &&
+                imports.find(tn) == imports.end() &&
                 tn != ls.structname.fullname) {
-                dependencies.insert(lm.type.nameUnderscore());
+                string lmPackage = topLevelPackage(lm.type.package);
+                if (lmPackage != "") {
+                    // The dependent type belongs to some package
+                    if (lmPackage == lsPackage) {
+                        // Then this is a dependency on a package from somewhere
+                        // in the same module, so we should import its outer
+                        // Julia module
+                        imports.insert("import _" + lm.type.nameUnderscore());
```

Comment:
Why not just import the types you need?

Reply:
Sorry, I think I don't understand what you mean. What would you propose?

Reply:
I guess I was confused as to why you have to import the whole module of your dependency rather than just the particular type you depend on.